connection.go

```diff
@@ -29,30 +29,40 @@ type StreamHandler func(stream *Stream)
 type AuthHandler func(header http.Header, slot uint8, parent uint32) bool
 
 type idleAwareFramer struct {
-	f       *spdy.Framer
-	conn    *Connection
-	expired *time.Timer
-	startCh chan struct{}
+	f              *spdy.Framer
+	conn           *Connection
+	expired        *time.Timer
```

Comment:
I take back the nil error, that was with `case <-i.expired.C` where i.expired was nil. I've added nil checks and your expireChan and all tests pass.